prost-build/src/code_generator.rs

```diff
@@ -531,37 +531,46 @@ impl <'a> CodeGenerator<'a> {
         let comments = Comments::from_location(self.location());
 
         self.path.push(2);
-        let methods = service.method
-                             .into_iter()
-                             .enumerate()
-                             .map(|(idx, mut method)| {
-                                 debug!("  method: {:?}", method.name());
-                                 self.path.push(idx as i32);
-                                 let comments = Comments::from_location(self.location());
-                                 self.path.pop();
-
-                                 let name = method.name.take().unwrap();
-                                 let input_proto_type = method.input_type.take().unwrap();
-                                 let output_proto_type = method.output_type.take().unwrap();
-                                 let input_type = self.resolve_ident(&input_proto_type);
-                                 let output_type = self.resolve_ident(&output_proto_type);
-                                 let client_streaming = method.client_streaming();
-                                 let server_streaming = method.server_streaming();
-
-                                 Method {
-                                     name: to_snake(&name),
-                                     proto_name: name,
-                                     comments,
-                                     input_type,
-                                     output_type,
-                                     input_proto_type,
-                                     output_proto_type,
-                                     options: method.options.unwrap_or_default(),
-                                     client_streaming,
-                                     server_streaming,
-                                 }
-                             })
-                             .collect();
+        let methods = service
+            .method
+            .into_iter()
+            .enumerate()
+            .map(|(idx, mut method)| {
+                debug!("  method: {:?}", method.name());
+                self.path.push(idx as i32);
+                let comments = Comments::from_location(self.location());
+                self.path.pop();
+
+                let name = method.name.take().unwrap();
+                let input_proto_type = method.input_type.take().unwrap();
+                let output_proto_type = method.output_type.take().unwrap();
+                let input_type = if let Some(ty) = self.known_type(&input_proto_type) {
```

Comment:
This is the relevant changes. The other lines of changes in this hunk is due to unintentionally triggered `rustfmt`.